src/wallet/test/wallet_tests.cpp

```diff
@@ -329,6 +329,39 @@ BOOST_FIXTURE_TEST_CASE(importwallet_rescan, TestChain100Setup)
     }
 }
 
+// This test verifies that wallet settings can be added and removed
+// Concurrently, ensuring no race conditions occur during either process.
+BOOST_FIXTURE_TEST_CASE(write_wallet_settings_concurrently, TestingSetup)
+{
+    WalletContext context;
+    context.chain = m_node.chain.get();
+    const int NUM_WALLETS = 5;
+    std::vector<std::thread> threads;
+    {
+        for (int i{0}; i < NUM_WALLETS; i++) {
+            threads.emplace_back([i, &context] {
+                BOOST_CHECK_MESSAGE(AddWalletSetting(context, strprintf("wallet_%d", i)), strprintf("write wallet_%d failed", i));
+            });
+        }
+        for (auto& t : threads)
+            t.join();
+        auto wallets = context.chain->getRwSetting("wallet");
+        BOOST_CHECK_EQUAL(wallets.getValues().size(), NUM_WALLETS);
+    }
+    threads.clear();
+    {
+        for (int i{0}; i < NUM_WALLETS; i++) {
+            threads.emplace_back([i, &context] {
+                BOOST_CHECK_MESSAGE(RemoveWalletSetting(context, strprintf("wallet_%d", i)), strprintf("write wallet_%d failed", i));
+            });
+        }
+        for (auto& t : threads)
+            t.join();
```

Comment:
style nit: should either use braces or multi-line, or move to one-line approach

```suggestion
        for (auto& t : threads) t.join();
```

Reply:
This is what was written initially. 
Clang format changed it to this format, which I thought was the recommended approach:
https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy

I've now switched to your suggestion.